homeassistant/core.py

```diff
@@ -408,6 +433,8 @@ def async_fire(self, event_type: str, event_data=None,
         for func in listeners:
```

Comment:
Oh, or maybe queue them in callbacks, then iterate through running them *after* adding the sync jobs to the thread-pool.